tests/test_butler.py

```diff
@@ -1372,6 +1372,15 @@ def testPytypeCoercion(self):
         datasetType_ori = butler.registry.getDatasetType(datasetTypeName)
         self.assertEqual(datasetType_ori.storageClass.name, "StructuredDataNoComponents")
 
+        # Put a dict and this should coerce to a MetricsExample
+        test_dict = {"summary": {"a": 1}, "output": {"b": 2}}
+        metric_ref = butler.put(test_dict, datasetTypeName, dataId=dataId, visit=424)
+        test_metric = butler.getDirect(metric_ref)
+        self.assertEqual(test_metric.summary, test_dict["summary"])
+        self.assertEqual(test_metric.output, test_dict["output"])
+        # Remove the dataset because another will be stored later in the test.
```

Comment:
I think it might be better to just duplicate the setup code and put this in another test method; it's incidental duplication (not so bad at all) vs. starting the kind of test daisy-chaining we've both regretted in the past, over in `runPutGetTests`.

Reply:
The test is called `testPytypeCoercion` so it's not a completely irrelevant thing to put in here. If I put it into a different run collection I wouldn't need to remove the dataset.

I do agree that we've made life hard for ourselves with `runPutGetTests` and then having every single datastore run every test.

Reply:
It's not the relevance of this to the test method that worries me; it's making one test's setup involve another test's cleanup - it makes it hard to modify the middle test without digging into the next one.

Reply:
I've separated it out and also factored out the butler creation code from runPutGetTest so we don't need to run all the tests again.